File: src/types/PersonaConfig.ts
import type { DirectorNotes } from './directorNotes';

export type PersonaConfig = CustomPersonaConfig;

export interface CustomPersonaConfig {
  personaId: string;
  name: string;
  avatarId: string;
  voiceId: string;
  llmId?: string;
  systemPrompt?: string;
  maxSessionLengthSeconds?: number;
  languageCode?: string;
  /**
   * Director Notes for the session — a baseline performance style for the
   * avatar and how expressively it is played. Cara 4 avatars only; forwarded
   * unchanged to session-token creation. See {@link DirectorNotes}.
   */
  directorNotes?: DirectorNotes;
}

// Confirmation that a runtime persona config update was applied.
export interface PersonaConfigUpdateAppliedEvent {
  // Changed config path -> { before, after } values.
  changedFields: Record<string, { before?: unknown; after?: unknown }>;
}

export function isCustomPersonaConfig(
  personaConfig: PersonaConfig,
): personaConfig is CustomPersonaConfig {
  return 'brainType' in personaConfig || 'llmId' in personaConfig;
}

File: flutter/lib/src/types/persona_config.dart
class PersonaConfig {
  final String? personaId;
  final String name;
  final String avatarId;
  final String voiceId;
  final String brainType;
  final String? systemPrompt;
  final int? maxSessionLengthSeconds;
  final String? languageCode;

  const PersonaConfig({
    this.personaId,
    required this.name,
    required this.avatarId,
    required this.voiceId,
    required this.brainType,
    this.systemPrompt,
    this.maxSessionLengthSeconds,
    this.languageCode,
  });

  Map<String, dynamic> toJson() {
    return {
      if (personaId != null) 'personaId': personaId,
      'name': name,
      'avatarId': avatarId,
      'voiceId': voiceId,
      'brainType': brainType,
      if (systemPrompt != null) 'systemPrompt': systemPrompt,
      if (maxSessionLengthSeconds != null)
        'maxSessionLengthSeconds': maxSessionLengthSeconds,
      if (languageCode != null) 'languageCode': languageCode,
    };
  }
}